service/src/test/java/com/codeforcommunity/processor/PostsProcessorTest.java

```diff
@@ -0,0 +1,352 @@
+ */
+public class PostsProcessorTest {
+  // We're using the mock table instance rather than the interface since we want access to the
+  // methods created for testing.
+  private MockCommentTable commentTable;
+  private MockPostTable postTable;
+  private PostsProcessor processor;
+
+  /**
+   * The {@code @BeforeEach} annotation lets JUnit know to run this method before running each test.
+   * This lets us clear anything that may have been in the database or changed in the previous
+   * processor so that we know each test will be run from the same starting point.
+   */
+  @BeforeEach
+  public void setup() {
+    // Setup a new CommentTable seeded with STUB_POST_COUNT posts.
+    this.commentTable = new MockCommentTable(STUB_POST_COUNT);
+    // Setup a new PostTable.
+    this.postTable = new MockPostTable(STUB_POST_COUNT);
+    // Setup a new PostsProcessor.
+    this.processor = new PostsProcessor(this.postTable, this.commentTable);
+  }
+
+  @Test
+  public void testGetPosts() {
+    // Get the posts from the processor.
+    PostsResponse posts = processor.getPosts();
+    // Make sure we have STUB_POST_COUNT posts.
+    assertEquals(STUB_POST_COUNT, posts.getPosts().size());
+
+    // Check each post to make sure it has properties we expect.
+    // Mainly that no fields are null, the date isn't blank, and integers are 0 or positive.
+    for (SinglePostResponse post : posts.getPosts()) {
+      assertNotNull(post.getId());
+      assertTrue(post.getId() >= 0);
+      assertNotNull(post.getAuthor());
+      assertNotNull(post.getTitle());
+      assertNotNull(post.getBody());
+      assertNotNull(post.getClapCount());
+      assertTrue(post.getClapCount() >= 0);
+      assertNotNull(post.getDateCreated());
+      assertFalse(post.getDateCreated().isBlank());
+      assertNotNull(post.getPreview());
+    }
+  }
+
+  @Test
+  public void testGetComments() {
+    // For each post that (should) exists.
+    for (int i = 0; i < STUB_POST_COUNT; i++) {
+      // Get the posts from the processor.
+      CommentsResponse comments = processor.getCommentsForPost(i);
+
+      // Check each post to make sure it has properties we expect.
+      // Mainly that no fields are null, the date isn't blank, and integers are 0 or positive.
+      // The postId should also be equal to the current post.
+      for (Comment comment : comments.getComments()) {
+        assertNotNull(comment.getId());
+        assertTrue(comment.getId() >= 0);
+        assertEquals(i, comment.getPostId());
+        assertNotNull(comment.getAuthor());
+        assertNotNull(comment.getBody());
+        assertNotNull(comment.getClapCount());
+        assertTrue(comment.getClapCount() >= 0);
+        assertNotNull(comment.getDateCreated());
+        assertFalse(comment.getDateCreated().isBlank());
+      }
+    }
+  }
+
+  // Write a test which runs multiple times with different inputs.
+  @ParameterizedTest
+  @ValueSource(ints = {1, 2, 5, 14})
```

Comment:
Do you describe what these annotations are anywhere?

Reply:
++ on that. I think it'll be nice to explain what `@ValueSource(ints = {-1, STUB_POST_COUNT + 1})` mean as well!